### ISSUE TITLE
"Import" assist places new "use" after the last existing one even if it's burried in the middle of a program

### ISSUE THREAD

**Issue by Vlad-Shcherbina** [rust-analyzer/rust-analyzer#5795]
To reproduce, apply "Import HashMap" assist to the following snippet:
```rust
enum Coin {
    Heads,
    Tails,
}
use Coin::*;

fn main() {
    let h = HashMap<|>::new();
}
```

Expected result:
```rust
use std::collections::HashMap;

enum Coin {
...
```

Actual result:
```rust
enum Coin {
    Heads,
    Tails,
}
use Coin::*;
use std::collections::HashMap;
...
```

**Comment by matklad** [comment]
Part of the giant yak shave: https://github.com/rust-analyzer/rust-analyzer/issues/3947

**Comment by Veykril** [comment]
Has been fixed in #5935.
